src/service/xhr-impl.js

```diff
@@ -125,43 +122,41 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
-    if (!getMode().test &&
-        this.ampdocSingle_ &&
-        Math.random() < 0.01 &&
-        parseUrl(input).origin != this.win.location.origin &&
-        !Services.viewerForDoc(this.ampdocSingle_).hasBeenVisible()) {
-      dev().error('XHR', 'attempted to fetch %s before viewer was visible',
-          input);
-    }
-
+    // The isSingleDoc check is required because if in shadow mode, this will
+    // throw a console error because the shellShadowDoc_ is not set when
+    // fetching the amp doc. So either the test-bind-impl or test pre setup in
+    // shadow mode tests needs to be fixed or there is a bug in ampdoc impl
+    // getAmpDoc.
+    // TODO(alabiaga): This should be investigated and fixed.
+    const ampdoc = this.ampdocService.isSingleDoc() ?
+      this.ampdocService.getAmpDoc(this.win.document) : null;
```

Comment:
Done.  The reason I added it was because I was trying to do away with the isSingleDoc() check..and just make the call getAmpDoc..which was then causing test-bind-impl.js to fail and I was trying to get it to work by passing in the node, though it was not sufficient as the AmpDoc.shellShadowDoc_ was not defined. I wrote a TODO on that line of code as it seems like buggy code to me..or the test is not set up properly. So i believe this is where your initial comment of why this is still required now makes sense...i had to actually debug through the code for it to actually make sense.